src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -161,15 +161,14 @@ public AgentSettings LoadSettings()
 
             // Loop getting agent name and pool name
             string poolName = null;
-            int poolId = 0;
-            string agentName = null;
+            AgentSettings agentSettings;
             WriteSection(StringUtil.Loc("RegisterAgentSectionHeader"));
 
             while (true)
             {
                 try
                 {
-                    poolId = await agentProvider.GetPoolId(command);
+                    agentSettings = await agentProvider.GetPoolId(command);
```

Comment:
just pass agentsetting to GetPoolId(), and modify agentSetting within provider, then don't have to return anything.